### PR TITLE
Update issue templates

### PR DESCRIPTION
**Pull request by YoDaMa**

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+assignees: ''
+
+---
+
+------------------------------- delete below -------------------------------
+
+INSTRUCTIONS
+==========
+
+If you do not follow the below instructions, your issue may be dismissed.
```
Comment (yzhong94) [inline .github/ISSUE_TEMPLATE/bug-report.md]: Maybe reword this a little, we want to be nice to our customers

Comment (yzhong94) [reply .github/ISSUE_TEMPLATE/bug-report.md]: Something like: we may be forced to close your issues to keep our issue quality high.

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+1. Search existing issues to avoid creating duplicates.
+
+2. If possible test using the latest release to make sure your issues has not already been fixed: 
+https://github.com/Azure/azure-iot-sdk-c/releases/latest 
+
+3. Do not share information from your Azure subscription here (connection strings, service names (IoTHub, Provisioning), etc...). Do so through your Azure subscription.
```
Comment (yzhong94) [inline .github/ISSUE_TEMPLATE/bug-report.md]: I'd clarify service names, maybe reword that to "IoT Hub name" and "Device Provisioning Service scope ID".

And instead of "do so through your Azure subscription", please use this "Please submit a ticket and get assistance from Microsoft support team using the following link:
https://docs.microsoft.com/en-us/azure/azure-supportability/how-to-create-azure-support-request"

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+For the MCVE, we recommend you If you are unsure how to enable logging, refer to this document: https://github.com/Azure/azure-iot-sdk-c/blob/master/doc/Iothub_sdk_options.md
+
+5. Delete these instructions before submitting the bug.
+
+------------------------------- delete above -------------------------------
+
+
+**Environment**
+Raspberry Pi, Raspbian Stretch Lite (Release 2018-11-13)
```
Comment (yzhong94) [inline .github/ISSUE_TEMPLATE/bug-report.md]: Should we specify OS, toolchain etc instead of calling it environment?

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+------------------------------- delete below -------------------------------
+
+Thank you for submitting a feature request! A few things:
+
+- Please check that a similar feature request has not already been submitted. Adding support to a currently existing feature request by commenting will increase the visibility and importance of your request beyond submitting a duplicate request. 
+
+- If this is a large feature request, please submit it to user voice.
+https://feedback.azure.com/forums/321918-azure-iot
+
+As an open source project, we always welcome PRs for new features. So if you know what needs to be done, feel free to create the change and send it our way.
```
Comment (yzhong94) [inline .github/ISSUE_TEMPLATE/feature-request.md]: We should be explicit.  Customers should create an issue describing the proposed solution before submitting a PR.  The last thing we want to do is reject a community PR :(

Comment (jspaith) [reply .github/ISSUE_TEMPLATE/feature-request.md]: How would a customer determine whether "this is a large feature request" or not?  Could we give guidance, like "if you know this request requires service changes" then point them to the feedback.azure?

Comment (jspaith) [inline .github/ISSUE_TEMPLATE/feature-request.md]: We should provide best practices for submitting PR's.  In addition to Yi's suggestion, the code will require unit testing, potentially e2e tests, and potentially documentation changes.  If the customer has any questions on setting this up or what we think set of tests should be, they should ask when opening the issue with proposed solution.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+---
+name: Feature request
```
Comment (jspaith) [inline .github/ISSUE_TEMPLATE/feature_request.md]: We have a file named feature_request.md (underscore) and feature-request.md (hyphenated).  Do we really need both??

File: .github/ISSUE_TEMPLATE/feature-request.md (hunk 4 above)
Comment (jspaith) [reply .github/ISSUE_TEMPLATE/feature-request.md]: Also they should follow coding conventions in the file(s) they're modifying when doing a pr.

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 3 above)
Comment (YoDaMa) [reply .github/ISSUE_TEMPLATE/bug-report.md]: Yes that could be good. What do you think would be a relevant header for this one? Knowing what device they are running on and which OS they are running the SDK on are the first two relevant details that jump out at me. Should there be any other?

File: .github/ISSUE_TEMPLATE/feature-request.md (hunk 4 above)
Comment (YoDaMa) [reply .github/ISSUE_TEMPLATE/feature-request.md]: I've edited this line to reflect asking them first to submit an issue so we can guide them on best practices for a PR.

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 3 above)
Comment (yzhong94) [reply .github/ISSUE_TEMPLATE/bug-report.md]: The development machine, the OS of the device, compiler

File: .github/ISSUE_TEMPLATE/feature-request.md (hunk 4 above)
Comment (yzhong94) [reply .github/ISSUE_TEMPLATE/feature-request.md]: Actually, customers may not even know what features require service change (e.g. Java SDK missing X.509 CA).  The guidance should be: if this feature requires implementation in multiple SDKs, consider using user voice.

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+==========
+
+If the below instructions are not followed, we may be forced to close your issue.
+
+1. Search existing issues to avoid creating duplicates.
+
+2. If possible test using the latest release to make sure your issues has not already been fixed: 
+https://github.com/Azure/azure-iot-sdk-c/releases/latest 
+
+3. Do not share information from your Azure subscription here (connection strings, service names (IoT Hub name, Device Provisioning Service scope ID), etc...). Please submit a ticket and get assistance from the Microsoft support team using the following link:
```
Comment (jspaith) [inline .github/ISSUE_TEMPLATE/bug-report.md]: >Please submit a ticket and get assistance from the Microsoft support team using the following link:  [](start = 160, length = 100)

"Please submit a ticket and get assistance from the Microsoft support team using the following link: "

Literally following this I'd always open the link :).  Could we change this to "If you need to share any of this information, please submit ..."

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+2. If possible test using the latest release to make sure your issues has not already been fixed: 
+https://github.com/Azure/azure-iot-sdk-c/releases/latest 
+
+3. Do not share information from your Azure subscription here (connection strings, service names (IoT Hub name, Device Provisioning Service scope ID), etc...). Please submit a ticket and get assistance from the Microsoft support team using the following link: 
+https://docs.microsoft.com/en-us/azure/azure-supportability/how-to-create-azure-support-request
+
+4. Include enough information for us to address the bug, including a description of the bug, a Minimal Complete Reproducible Example that demonstrates the bug, and any relevant logs related to the bug.  
+
+For the MCVE, we recommend you provide us with a code snippet we can cut and paste into a readily available sample, or a link to a code sample that we can compile to reproduce the bug.
```
Comment (jspaith) [inline .github/ISSUE_TEMPLATE/bug-report.md]: >MCVE [](start = 8, length = 4)

MCVE - please put the link to this concept [https://stackoverflow.com/help/mcve](https://stackoverflow.com/help/mcve) 1st time that you use it.

Comment (jspaith) [reply .github/ISSUE_TEMPLATE/bug-report.md]: Or else put an MCVE in parens and link ideally next to the Minimal... the 1st time you spell out the full concept in sentence above.

---
In reply to: [263177036](https://github.com/Azure/azure-iot-sdk-c/pull/886#discussion_r263177036) [](ancestors = 263177036)

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+4. Include enough information for us to address the bug, including a description of the bug, a Minimal Complete Reproducible Example that demonstrates the bug, and any relevant logs related to the bug.  
+
+For the MCVE, we recommend you provide us with a code snippet we can cut and paste into a readily available sample, or a link to a code sample that we can compile to reproduce the bug. 
+
+If you are unsure how to enable logging, refer to this document: https://github.com/Azure/azure-iot-sdk-c/blob/master/doc/Iothub_sdk_options.md
+
+5. Delete these instructions before submitting the bug.
+
+------------------------------- delete above -------------------------------
+
```
Comment (jspaith) [inline .github/ISSUE_TEMPLATE/bug-report.md]: For text below this, could we make it more clear that this is an sample of a good hypothetical report that we could take action on.

File: .github/ISSUE_TEMPLATE/technical-question.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+labels: question
+assignees: ''
+
+---
+
+------------------------------- delete below -------------------------------
+
+The best place to submit technical questions related to the SDK is on Stack Overflow. Simply submit a question with the tag "azure-iot-hub". 
+
+Need Technical Support? Every customer with an active Azure subscription has access to support with guaranteed response time. Consider submitting a ticket and get assistance from Microsoft support team using the following link:
```
Comment (jspaith) [inline .github/ISSUE_TEMPLATE/technical-question.md]: >Need Technical Support? [](start = 0, length = 23)

It's still not clear when I'd want to use the "Technical questions" of line above versus "Need technical support"?  Might want to check with Yi if there's more precise guidanec around this?

If possible it's best if folks can hit up public forums, if only so there's an ongoing record of discussion and other people can search it in future.

Comment (YoDaMa) [reply .github/ISSUE_TEMPLATE/technical-question.md]: This is a good point of clarification. Let's let Yi weigh in.

Comment (YoDaMa) [reply .github/ISSUE_TEMPLATE/technical-question.md]: I've removed this from the technical-question doc since it isn't relevant.

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+assignees: ''
+
+---
+
+------------------------------- delete below -------------------------------
+
+INSTRUCTIONS
+==========
+
+Please follow the instructions and template below. Otherwise, to maintain quality of issues we may be forced to close your issue.
```
Comment (jspaith) [inline .github/ISSUE_TEMPLATE/bug-report.md]: >Otherwise, to maintain quality of issues we may be forced to close your issue. [](start = 51, length = 78)

I can't imagine us closing an issue on someone in practice.  Could we reword the "Otherwise, to maintain quality...' to something like

"Otherwise, we may have to spend additional time requesting this information from you and your resolution time may be delayed."
